transformations/sentiment_emoji_augmenter/transformation.py

```diff
@@ -0,0 +1,115 @@
+import random
+
+from interfaces.SentenceOperation import SentenceAndTargetOperation
+import spacy
+from tasks.TaskTypes import TaskType
+
+"""
+Adds a positive labelled emoji as well as a positive emoteicon for positive sentences and vice versa. 
+And neutral smiley for unlabelled and neutral sentences.
+Since IMDB has labels +1 --> str(target) in ["1", "pos", "positive"] is used to determine if it's positive.
+"""
+
+emoji = {  # (facial expression, sentiment)-keys
+    ("love", +1.00): ["❤️", "💜", "💚", "💙", "💛", "💕"],
+    ("grin", +1.00): ["😀", "😄", "😃", "😆", "😅", "😂", "😁", "😻", "😍", "😈", "👌"],
+    ("taunt", +0.75): ["😛", "😝", "😜", "😋", "😇"],
+    ("smile", +0.50): ["😊", "😌", "😏", "😎", "☺", "👍"],
+    ("wink", +0.25): ["😉"],
+    ("blank", +0.00): ["😐", "😶"],
+    ("gasp", -0.05): ["😳", "😮", "😯", "😧", "😦", "🙀"],
+    ("worry", -0.25): ["😕", "😬"],
+    ("frown", -0.75): ["😟", "😒", "😔", "😞", "😠", "😩", "😫", "😡", "👿"],
+    ("cry", -1.00): ["😢", "😥", "😓", "😪", "😭", "😿"],
+}
+
+emoticons = {  # (facial expression, sentiment)-keys
+    ("love", +1.00): ["<3", "♥", "❤"],
+    ("grin", +1.00): [">:D", ":-D", ":D", "=-D", "=D", "X-D", "x-D", "XD", "xD", "8-D"],
+    ("taunt", +0.75): [">:P", ":-P", ":P", ":-p", ":p", ":-b", ":b", ":c)", ":o)", ":^)"],
+    ("smile", +0.50): [">:)", ":-)", ":)", "=)", "=]", ":]", ":}", ":>", ":3", "8)", "8-)"],
+    ("wink", +0.25): [">;]", ";-)", ";)", ";-]", ";]", ";D", ";^)", "*-)", "*)"],
+    ("blank", +0.00): [":-|", ":|"],
+    ("gasp", -0.05): [">:o", ":-O", ":O", ":o", ":-o", "o_O", "o.O", "°O°", "°o°"],
+    ("worry", -0.25): [">:/", ":-/", ":/", ":\\", ">:\\", ":-.", ":-s", ":s", ":S", ":-S", ">.>"],
+    ("frown", -0.75): [">:[", ":-(", ":(", "=(", ":-[", ":[", ":{", ":-<", ":c", ":-c", "=/"],
+    ("cry", -1.00): [":'(", ":'''(", ";'("]
+}
+
+positive_emojis = [k for k in emoji.keys() if k[1] > 0.2]
+neutral_emojis = [k for k in emoji.keys() if 0.2 > k[1] > -0.2]
+negative_emojis = [k for k in emoji.keys() if k[1] < -0.2]
+
+
+class SentimentEmojiAugmenter(SentenceAndTargetOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.SENTIMENT_ANALYSIS]
+    languages = "All"
+    tgt_languages = "All"
+
+    def __init__(self, seed=0):
+        super().__init__(seed)
+        self.nlp = spacy.load("en_core_web_sm")
+        self.seed = seed
+        random.seed(self.seed)
+
+    def generate(self, sentence: str, target: str):
+        if target is None:
+            emotions = self.get_emotions("neutral")
+        elif str(target) in ["1", "pos", "positive"]:
+            emotions = self.get_emotions("pos")
+        elif str(target) in ["0", "neg", "negative"]:
+            emotions = self.get_emotions("neg")
+        else:
+            emotions = self.get_emotions("neutral")
+        perturbed_sentences = [sentence + " " + emotion for emotion in emotions if emotion.strip()]
```

Comment:
This is limiting emojis to end of sequences. Since you are already using spacy, could you insert them after every sentence (probabilistically)?